src/ui/webeiddialog.cpp

```diff
@@ -399,20 +437,28 @@ bool WebEidDialog::event(QEvent* event)
 {
     if (event->type() == QEvent::LanguageChange) {
         ui->retranslateUi(this);
-        ui->aboutVersion->setText(tr("Version: %1").arg(qApp->applicationVersion()));
+        emit languageChange();
```

Comment:
Qt [docs say](https://doc.qt.io/qt-5/signalsandslots.html#signals):

> When a signal is emitted, the slots connected to it are usually executed immediately, just like a normal function call. When this happens, the signals and slots mechanism is totally independent of any GUI event loop. Execution of the code following the emit statement will occur once all slots have returned. The situation is slightly different when using [queued connections](https://doc.qt.io/qt-5/qt.html#ConnectionType-enum); in such a case, the code following the emit keyword will continue immediately, and the slots will be executed later.

As far as I can see, this `emit` does not use queued connections and so the `languageChange` events of all the widgets will be fully processed after this line. In this case we can happily redraw the dialog here, knowing that the new size of everything has been calculated properly:

```suggestion
        emit languageChange();
        resizeHeight();
```